ast/compile.go

```diff
@@ -260,6 +262,7 @@ func NewCompiler() *Compiler {
 	}{
 		{"CheckDuplicateImports", "compile_stage_check_duplicate_imports", c.checkDuplicateImports},
 		{"CheckKeywordOverrides", "compile_stage_check_keyword_overrides", c.checkKeywordOverrides},
+		{"SetSelfUsage", "compile_stage_set_self_usage", c.setSelfUsage},
 		// Reference resolution should run first as it may be used to lazily
 		// load additional modules. If any stages run before resolution, they
```

Comment:
☝️ I still don't fully understand _when_ that is the case: When are modules lazily loaded during reference resolution?

For this issue, could they not include `self.metadata.*()` references, and we would not rewrite those because `SetSelfUsage` would not mark them?

Another option (which I had also considered when working on the _unused imports_ stuff) would to simply _keep_ the imports in `ResolveRefs`. So what's `ResolveRefs` now would become

- `ResolveRefs` (keeping imports)
- [...] whatever stage that needs the imports
- `RemoveImports` which would kill all imports.